data_pipeline/kubernetes/kubernetes_pipeline_config.py

```diff
@@ -36,41 +60,75 @@ class KubernetesPipelineConfig:  # pylint: disable=too-many-instance-attributes
 
     @staticmethod
     def from_dict(
-        pipeline_config_dict: KubernetesPipelineConfigDict
+        pipeline_config_dict: KubernetesPipelineConfigDict,
+        default_config_dict: Optional[KubernetesDefaultConfigDict] = None
     ) -> 'KubernetesPipelineConfig':
+        default_config_dict = default_config_dict or {}
         return KubernetesPipelineConfig(
             data_pipeline_id=pipeline_config_dict['dataPipelineId'],
-            airflow_config=AirflowConfig.from_optional_dict(pipeline_config_dict.get('airflow')),
+            airflow_config=AirflowConfig.from_optional_dict(
+                airflow_config_dict=pipeline_config_dict.get('airflow'),
+                default_airflow_config=AirflowConfig.from_optional_dict(
+                    default_config_dict.get('airflow')
+                )
+            ),
             image=pipeline_config_dict['image'],
             arguments=pipeline_config_dict['arguments'],
             volume_mounts=[
                 convert_dict_to_kubernetes_client_object(
                     config_dict,
                     k8s_models.V1VolumeMount
                 )
-                for config_dict in pipeline_config_dict['volumeMounts']
+                for config_dict in get_merged_config_dict_list(
+                    config=pipeline_config_dict.get('volumeMounts', []),
+                    default_config=default_config_dict.get('volumeMounts', []),
+                    unique_keys=['name', 'mountPath']
```

Comment:
I believe the key is `mountPath` only. Because you can't mount more than one thing to the same path.

Reply:
same name but different mountPath is ok then?

Reply:
> same name but different mountPath is ok then?

Yes, I believe so. Which on its own may not be useful. But using the `subPath` would allow you to mount a select sub path from the resource specified by `name` to the `mountPath`.